src/master-playlist-controller.js

```diff
@@ -1366,8 +1366,15 @@ export class MasterPlaylistController extends videojs.EventTarget {
       let variantCodecCount = 2;
       const blacklistReasons = [];
 
-      if (variant.attributes.CODECS) {
-        variantCodecs = parseCodecs(variant.attributes.CODECS);
+      // get codecs from the playlist for this variant
+      const playlistCodecs = codecsForPlaylist(this.masterPlaylistLoader_.master, variant);
```

Comment:
I think it may be worth considering a different name here. `playlistCodecs` and `variantCodecs` are essentially the same thing, but represent different objects. Maybe something like `variantCodecStrings` here, or something else.